Rules/NoConfigurationBlock.cs

```diff
@@ -0,0 +1,99 @@
+﻿//
+// Copyright (c) Microsoft Corporation.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.
+//
+
+using System;
+using System.Collections.Generic;
+using System.Management.Automation.Language;
+using Microsoft.Windows.Powershell.ScriptAnalyzer.Generic;
+using System.ComponentModel.Composition;
+using System.Globalization;
+
+namespace Microsoft.Windows.Powershell.ScriptAnalyzer.BuiltinRules
+{
+    /// <summary>
+    /// NoConfigurationBlock: Check if any configuration block is used. Workflow is not supported on NanoServer.
+    /// </summary>
+    [Export(typeof(IScriptRule))]
+    public class NoConfigurationBlock : IScriptRule
+    {
+        /// <summary>
+        /// AnalyzeScript: Check if workflow block is used.
+        /// </summary>
+        public IEnumerable<DiagnosticRecord> AnalyzeScript(Ast ast, string fileName)
+        {
+            if (ast == null) throw new ArgumentNullException(Strings.NullAstErrorMessage);
+            IEnumerable<Ast> funcs = ast.FindAll(testAst => testAst is FunctionDefinitionAst, true);
+            foreach (FunctionDefinitionAst func in funcs)
+            {
+                if (String.Equals(func.Name,"Configuration",StringComparison.CurrentCultureIgnoreCase))
```

Comment:
Actually, in PowerShell 5 this is ConfigurationDefinitionAst. I tested on my console and it seems that it will not show up as a FunctionDefinitionAst.